packages/@sanity/cli/src/util/remoteTemplate.ts

```diff
@@ -0,0 +1,508 @@
+      if (!hasDataset) missing.push('SANITY_DATASET or SANITY_STUDIO_DATASET')
+      throw new Error(
+        `Environment template in ${
+          packagePath || 'repo'
+        } must include the following variables: ${missing.join(', ')}`,
+      )
+    }
+  }
+
+  return {
+    hasSanityConfig,
+    hasSanityCli,
+    hasEnvFile: Boolean(envFile),
+    hasSanityDep,
+  }
+}
+
+/**
+ * Validates a GitHub repository template against required criteria.
+ * Supports both monorepo and single-package repositories.
+ *
+ * For monorepos:
+ * - Each package must have a valid package.json
+ * - At least one package must include 'sanity' in dependencies or devDependencies
+ * - At least one package must have sanity.config.js/ts and sanity.cli.js/ts
+ * - Each package must have a .env.template, .env.example, or .env.local.example
+ *
+ * For single-package repositories:
+ * - Must have a valid package.json with 'sanity' dependency
+ * - Must have sanity.config.js/ts and sanity.cli.js/ts
+ * - Must have .env.template, .env.example, or .env.local.example
+ *
+ * Environment files must include:
+ * - SANITY_PROJECT_ID or SANITY_STUDIO_PROJECT_ID variable
+ * - SANITY_DATASET or SANITY_STUDIO_DATASET variable
+ *
+ * @throws Error if validation fails with specific reason
+ */
+export async function validateRemoteTemplate(
+  repoInfo: RepoInfo,
+  packages: string[] = [''],
+  bearerToken?: string,
+): Promise<void> {
+  const {username, name, branch, filePath} = repoInfo
+  const baseUrl = `https://raw.githubusercontent.com/${username}/${name}/${branch}/${filePath}`
+
+  const headers: Record<string, string> = {}
+  if (bearerToken) {
+    headers.Authorization = `Bearer ${bearerToken}`
+  }
+
+  const validations = await Promise.all(
+    packages.map((pkg) => validatePackage(baseUrl, pkg, headers, pkg === '')),
+  )
+
+  const hasSanityDep = validations.some((v) => v.hasSanityDep)
+  if (!hasSanityDep) {
+    throw new Error('At least one package must include "sanity" as a dependency in package.json')
+  }
+
+  const hasSanityConfig = validations.some((v) => v.hasSanityConfig)
+  if (!hasSanityConfig) {
+    throw new Error('At least one package must include a sanity.config.js or sanity.config.ts file')
+  }
+
+  const hasSanityCli = validations.some((v) => v.hasSanityCli)
+  if (!hasSanityCli) {
+    throw new Error('At least one package must include a sanity.cli.js or sanity.cli.ts file')
+  }
+
+  const missingEnvPackages = packages.filter((pkg, i) => !validations[i].hasEnvFile)
+  if (missingEnvPackages.length > 0) {
+    throw new Error(
+      `The following packages are missing .env.template, .env.example, or .env.local.example files: ${missingEnvPackages.join(
+        ', ',
+      )}`,
+    )
+  }
+}
+
+export async function isNextJsTemplate(root: string): Promise<boolean> {
+  try {
+    const packageJson = await readFile(join(root, 'package.json'), 'utf8')
```

Comment:
not a huge deal I guess, but would prefer if we read and parsed `package.json` once and instead passed the parsed value to the functions that need it

Reply:
Did a small test, and I don't think it really matters in this case. 

Reading package.json 1000 times...
Average time: 0.0585ms

Personally I much rather prefer the logic being encapsulated within the functions. Wdyt?

Reply:
Sure, I can see why that makes sense!